src/flamenco/vm/fd_vm_private.h

```diff
@@ -100,17 +102,103 @@ typedef struct fd_vm_vec fd_vm_vec_t;
 #define FD_VM_SBPF_ENABLE_NEG(v)                   ( v <  FD_SBPF_V2 )
 #define FD_VM_SBPF_SWAP_SUB_REG_IMM_OPERANDS(v)    ( v >= FD_SBPF_V2 )
 #define FD_VM_SBPF_EXPLICIT_SIGN_EXT(v)            ( v >= FD_SBPF_V2 )
-/* SIMD-0176 */
+/* SIMD-0178 + SIMD-0179 */
 #define FD_VM_SBPF_STATIC_SYSCALLS(v)              ( v >= FD_SBPF_V3 )
-/* SIMD-XXXX */
-#define FD_VM_SBPF_STRICTER_CONTROLFLOW(v)         ( v >= FD_SBPF_V3 )
-#define FD_VM_SBPF_REJECT_RODATA_STACK_OVERLAP(v)  ( v >= FD_SBPF_V3 )
-#define FD_VM_SBPF_ENABLE_ELF_VADDR(v)             ( v >= FD_SBPF_V3 )
+/* SIMD-0189 */
+#define FD_VM_SBPF_ENABLE_STRICTER_ELF_HEADERS(v)  ( v >= FD_SBPF_V3 )
+#define FD_VM_SBPF_ENABLE_LOWER_BYTECODE_VADDR(v)  ( v >= FD_SBPF_V3 )
 
 #define FD_VM_SBPF_DYNAMIC_STACK_FRAMES_ALIGN      (64U)
 
 #define FD_VM_OFFSET_MASK (0xffffffffUL)
 
+static const uint FD_VM_SBPF_STATIC_SYSCALLS_LIST[] = {
```

Comment:
I know these are the fd_murmur3_32 of the syscall names... Does this table exist some place in solana as well?

Reply:
in theory, in v3 we no longer need a hash map, we just need an array, because the indexes go from a hash to a sequential id.
the complexity is that we have to track what's active vs not.

two options:
1. create an array, set active/disabled syscalls inside that array, look up the syscall in the array
2. create an array that we use to lookup inside the existing map (where activation is already handled)

1. is one lookup faster, but requires us to keep in sync both map and array, at least until we have to support anything before v3. 2. has the advantage that there's a single place that's the source of truth for activation, at the price of a dbl lookup.

I chose 2 because v3 is pretty far away in the future, and v0..v2 will be in use for a pretty long time even after v3 is active. makes sense?

(agave has their own function registry with a completely different internal impl.)